pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -157,6 +157,13 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		bestHint = extendedHint
 	}
 
+	// the best hint might violate the NUMA allocation rule on which
```

Comment:
is the `Preferred` flag relevant here or is the issue independent from it?
Reason I ask is that in your test you had a multi-zone CPU Preferred hint, which is in my experience rare (very rare I'd say). Most of the CPU hints I've seen spanning across NUMA zones are non-preferred (and for a good reason)

Reply:
> is the Preferred flag relevant here or is the issue independent from it?

The issue discovered in `best-effort` policy which accepts the hint even if it's not preferred so the flag doesn't matter in this case.

Actually in `restricted` we would probably won't hit this bug, exactly for the reason you mentioned.

Reply:
thanks, this is a very important information (which the issue emerged using the topology manager policy best-effort) that I somehow overlooked